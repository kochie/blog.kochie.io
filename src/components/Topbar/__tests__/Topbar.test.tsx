import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'
import Link from 'next/link'
import TopBar from '@/components/Topbar'

describe('TOPBAR COMPONENT', () => {
  test('should render', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<TopBar />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('should link to author', () => {
    const tree = create(<TopBar />)
    const links = tree.root.findAllByType(Link)
    expect(links[0].props['href']).toBe('/authors')
  })

  test('should link to root (articles)', () => {
    const tree = create(<TopBar />)
    const links = tree.root.findAllByType(Link)
    expect(links[1].props['href']).toBe('/')
  })

  test('should link to tags', () => {
    const tree = create(<TopBar />)
    const links = tree.root.findAllByType(Link)
    expect(links[2].props['href']).toBe('/tags')
  })
})

it('should link to author', () => {
  const tree = create(<TopBar />)
  const links = tree.root.findAllByType(Link)
  expect(links[0].props['href']).toBe('/authors')
})

it('should link to root (articles)', () => {
  const tree = create(<TopBar />)
  const links = tree.root.findAllByType(Link)
  expect(links[1].props['href']).toBe('/')
})

it('should link to tags', () => {
  const tree = create(<TopBar />)
  const links = tree.root.findAllByType(Link)
  expect(links[2].props['href']).toBe('/tags')
})
