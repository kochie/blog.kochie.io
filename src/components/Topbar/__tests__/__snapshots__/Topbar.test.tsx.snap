// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOPBAR COMPONENT should render 1`] = `
<div
  className="flex flex-row justify-center items-center list-none bg-gray-500 m-0 p-0 fixed w-screen z-40"
>
  <div
    className="child heading"
    href="/authors"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Authors
  </div>
  <div
    className="child heading"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Articles
  </div>
  <div
    className="child heading"
    href="/tags"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Tags
  </div>
</div>
`;
